### ISSUE TITLE
LSTM Cell Training Error

### ISSUE THREAD

**Issue by ilhan90** [tensorflow/tensorflow#23438]
<em>I have a problem with multi layer lstm training. The accuracy and output model slowly changing when calling "sess.run(logits, feed_dict={X:...})" during train process. The problem doesn't happens when I decreased number of layers or neurons or dataset size.

**System information**
- Google colab selecting gpu
- Python version 3

```
n_epochs = 6
batch_size = 150

n_layers = 3
n_neurons = 300
n_outputs = 2
learning_rate = 0.001


reset_graph()

X = tf.placeholder(tf.float32, [None, n_steps, n_inputs])
y = tf.placeholder(tf.int32, [None])
devices = ["/gpu:0"] # replace with ["/gpu:0", "/gpu:1", "/gpu:2"]
lstm_cells = [DeviceCellWrapper(dev,tf.contrib.rnn.LSTMCell(num_units=n_neurons))
       for layer in range(n_layers)
       for dev in devices]
multi_cell = tf.contrib.rnn.MultiRNNCell(lstm_cells)      
outputs, states = tf.nn.dynamic_rnn(multi_cell, X, dtype=tf.float32)
top_layer_h_state = states[-1][1]
logits = tf.layers.dense(top_layer_h_state, n_outputs, name="softmax")
xentropy = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=y, logits=logits)
loss = tf.reduce_mean(xentropy, name="loss")
optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate)
training_op = optimizer.minimize(loss)
correct = tf.nn.in_top_k(logits, y, 1)
accuracy = tf.reduce_mean(tf.cast(correct, tf.float32))

init = tf.global_variables_initializer()


saver = tf.train.Saver()


with tf.Session() as sess:
    init.run()
    maxValidationPoint = float('-inf')

    for epoch in range(n_epochs):
        trainDataIndexInPtr = 0
        while(True):
            X_batch, y_batch = next_batch(batch_size, X_train, y_train)
            sess.run(training_op, feed_dict={X: X_batch, y: y_batch})
            if(trainDataIndexInPtr == 0):
                break

        acc_train = accuracy.eval(feed_dict={X: X_train.reshape((len(X_train), n_steps, n_inputs)), y: y_train.reshape((len(y_train)))})
        acc_validation = accuracy.eval(feed_dict={X: X_validation, y: y_validation})

        y_pred = sess.run(logits, feed_dict={X: X_validation})
        y_pred2 = sess.run(logits, feed_dict={X: X_test})

        print("Train accuracy =", "%.7f" % acc_train, "Validation accuracy =", "%.7f" % acc_validation)

        if(acc_validation > maxAcc_validation ):
            maxAcc_validation = acc_validation
            saver.save(sess, "./Model")
```

### Output:

Train accuracy = 0.6400015 Validation accuracy = 0.6296040
Train accuracy = 0.6409340 Validation accuracy = 0.6253070
Train accuracy = 0.6552114 Validation accuracy = 0.6393685
Train accuracy = 0.6643262 Validation accuracy = 0.6265539
Train accuracy = 0.6804933 Validation accuracy = 0.6232543
Train accuracy = 0.7023602 Validation accuracy = 0.6231584

### if I throw away "y_pred = sess.run(logits, feed_dict={X: X_validation})" and "y_pred2 = sess.run(logits, feed_dict={X: X_test})", Output:

Train accuracy = 0.6400015 Validation accuracy = 0.6296040
Train accuracy = 0.6409340 Validation accuracy = 0.6253070
Train accuracy = 0.6552114 Validation accuracy = 0.6393685
Train accuracy = 0.6643139 Validation accuracy = 0.6265347
**Train accuracy = 0.6794249 Validation accuracy = 0.6231584
Train accuracy = 0.7062445 Validation accuracy = 0.6132405**

**Comment by harshini-gadige** [comment]
@drpngx   PTAL
